proposals/compatibility-mode.md

```diff
@@ -197,6 +197,18 @@ generate a validation error.
 other APIs only support the first vertex so only `@interpolation(flat, either)` is supported in
 compatibility mode.
 
+## 18. Introduce new `maxVertexStageStorageBuffers` and `maxVertexStageStorageTextures` limits.
+
+If the number of shader variables of type `storage_buffer` in a vertex shader exceeds the `maxVertexStageStorageBuffers` limit, a validation error will occur at pipeline creation time.
+
+If the number of shader variables of type `texture_storage_1d`, `texture_storage_2d`, `texture_storage_2d_array` and `texture_storage_3d` in a vertex shader exceeds the `maxVertexStageStorageTextures` limit, a validation error will occur at pipeline creation time.
+
+In Compatibility mode, these new limits will have a default of zero. In Core mode, they will default to the same values as the existing `maxStorageBuffersPerShaderStage` and `maxStorageTexturesPerShaderStage`, respectively.
```

Comment:
Ideas for how exactly this works mechanically in the spec:
- ~in core, just have the two new limits default to the same defaults as the old ones, while still enforcing them.~ This does not work because it would introduce new limits that need to be raised for core content (it's a breaking change).
- in core, skip validating the two new limits. This is technically fine but very unintuitive - you'd have some limits that you think you need to set, but don't.
- in requestDevice, "reify" the limits request by upgrading, e.g. `maxVertexStageStorageBuffers` so it's always at least as high as `maxStorageBuffersPerShaderStage`
  - (I know I've suggested similar things in the past for other pairs of limits, to make devices always match the [Adapter Capability Guarantees](https://gpuweb.github.io/gpuweb/#adapter-capability-guarantees) just to polish some corners, but I can't find the past discussion for the life of me. OTOH in this proposal it would actually be very important, to avoid being a breaking change.)
- in core, just have these two new limits default to infinity / a very large value, to signal that the limits don't exist. I prefer this one as I think it reduces surprises.

Reply:
I think I'd also prefer to have them default to `u32::MAX` in core with a note saying that the per stage limits still apply.